server/src/main/java/org/apache/iotdb/db/rest/OpenApiServer.java

```diff
@@ -0,0 +1,177 @@
+import org.eclipse.jetty.servlet.ServletHolder;
+import org.eclipse.jetty.util.ssl.SslContextFactory;
+import org.glassfish.jersey.servlet.ServletContainer;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import javax.servlet.DispatcherType;
+
+import java.util.EnumSet;
+
+public class OpenApiServer implements IService {
+  private static final Logger LOGGER = LoggerFactory.getLogger(OpenApiServer.class);
+
+  public static OpenApiServer getInstance() {
+    return OpenApiServerHolder.INSTANCE;
+  }
+
+  Server server;
+
+  private void start(int port) {
+    server = new Server(port);
+    ServletContextHandler context = getServletContextHandler();
+    server.setHandler(context);
+    serverStart();
+  }
+
+  private ServletContextHandler getServletContextHandler() {
+    ServletContextHandler context = new ServletContextHandler(ServletContextHandler.NO_SESSIONS);
+    context.addFilter(
+        ApiOriginFilter.class, "/*", EnumSet.of(DispatcherType.INCLUDE, DispatcherType.REQUEST));
+    ServletHolder holder = context.addServlet(ServletContainer.class, "/*");
+    holder.setInitOrder(1);
+    holder.setInitParameter(
+        "jersey.config.server.provider.packages",
+        "io.swagger.jaxrs.listing, io.swagger.sample.resource, org.apache.iotdb.db.rest");
+    holder.setInitParameter(
+        "jersey.config.server.provider.classnames",
+        "org.glassfish.jersey.media.multipart.MultiPartFeature");
+    holder.setInitParameter("jersey.config.server.wadl.disableWadl", "true");
+    context.setContextPath("/");
+    return context;
+  }
+
+  private void startSSL(
+      int port,
+      String keyStorePath,
+      String trustStorePath,
+      String keyStorePwd,
+      String trustStorePwd,
+      int idleTime) {
+    server = new Server();
+    HttpConfiguration https_config = new HttpConfiguration();
+    https_config.setSecurePort(port);
+    https_config.addCustomizer(new SecureRequestCustomizer());
+    SslContextFactory.Server sslContextFactory = new SslContextFactory.Server();
+    sslContextFactory.setKeyStorePath(keyStorePath);
+    sslContextFactory.setKeyStorePassword(keyStorePwd);
+    sslContextFactory.setTrustStorePath(trustStorePath);
+    sslContextFactory.setTrustStorePassword(trustStorePwd);
+    ServerConnector httpsConnector =
+        new ServerConnector(
+            server,
+            new SslConnectionFactory(sslContextFactory, HttpVersion.HTTP_1_1.asString()),
+            new HttpConnectionFactory(https_config));
+    httpsConnector.setPort(port);
+    httpsConnector.setIdleTimeout(idleTime);
+    server.addConnector(httpsConnector);
+    ServletContextHandler context = getServletContextHandler();
+    server.setHandler(context);
+    serverStart();
+  }
+
+  private void serverStart() {
+    try {
+      server.start();
+      server.join();
+    } catch (Exception e) {
+      LOGGER.warn("OpenApiServer start failed: {}", e.getMessage());
+    } finally {
```

Comment:
Throw the exception here and let the top-level class handle it. Terminating the start-up process is better when exceptions occur.